torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -818,6 +820,14 @@ class AsyncAllreduceWork : public ProcessGroupGloo::AsyncWork {
     GENERATE_ALL_TYPES(dtype, getFunction, fn, op);
     return fn;
   }
+
+
+  std::vector<at::Tensor> result() const override {
```

Comment:
What happens if the result() is not ready (such as operation was ran asynchronously?) We should probably have a check for this, otherwise there may be a risk of getting back incorrect results. 

We can probably do something like `TORCH_CHECK(isCompleted())` and throw an appropriate error message if it is not.

Reply:
That's a good point, done. PTAL.